src/framework/components/sound/slot.js

```diff
@@ -0,0 +1,573 @@
+            for (var i = 0, len = instances.length; i < len; i++) {
+                if (instances[i].stop())
+                    stopped = true;
+            }
+
+            instances.length = 0;
+
+            if (stopped) {
+                this.fire('stop', this);
+            }
+
+            return stopped;
+        },
+
+        /**
+         * @function
+         * @name  pc.SoundSlot#load
+         * @description Loads the asset assigned to this slot.
+         */
+        load: function () {
+            if (! this._hasAsset())
+                return;
+
+            var asset = this._assets.get(this._asset);
+            if (! asset) {
+                this._assets.off('add:' + this._asset, this._onAssetAdd, this);
+                this._assets.once('add:' + this._asset, this._onAssetAdd, this);
+                return;
+            }
+
+            asset.off('remove', this._onAssetRemoved, this);
+            asset.on('remove', this._onAssetRemoved, this);
+
+            if (!asset.resource) {
+                asset.off('load', this._onAssetLoad, this);
+                asset.once('load', this._onAssetLoad, this);
+
+                this._assets.load(asset);
+
+                return;
+            }
+
+            this.fire('load', asset.resource);
+        },
+
+        /**
+         * @function
+         * @name  pc.SoundSlot#setExternalNodes
+         * @description Connect external Web Audio API nodes. Any sound played by this slot will
+         * automatically attach the specified nodes to the source that plays the sound. You need to pass
+         * the first node of the node graph that you created externally and the last node of that graph. The first
+         * node will be connected to the audio source and the last node will be connected to the destination of the AudioContext (e.g speakers).
+         * @param {AudioNode} firstNode The first node that will be connected to the audio source of sound instances.
+         * @param {AudioNode} [lastNode] The last node that will be connected to the destination of the AudioContext.
+         * If unspecified then the firstNode will be connected to the destination instead.
+         * @example
+         * var context = app.systems.sound.context;
+         * var analyzer = context.createAnalyzer();
+         * var distortion = context.createWaveShaper();
+         * var filter = context.createBiquadFilter();
+         * analyzer.connect(distortion);
+         * distortion.connect(filter);
+         * slot.setExternalNodes(analyzer, filter);
+         */
+        setExternalNodes: function (firstNode, lastNode) {
+            if (! (firstNode)) {
+                logError('The firstNode must have a valid AudioNode');
+                return;
+            }
+
+            if (! lastNode) {
+                lastNode = firstNode;
+            }
+
+            this._firstNode = firstNode;
+            this._lastNode = lastNode;
+
+            // update instances if not overlapping
```

Comment:
There is no 'glitch' if there is it's a browser issue. For example there is no glitch in Firefox. Regarding your second question you can either set it to overlapping or you can subscibe to the 'stop' event and do it then.